.github/workflows/test.yml

```diff
@@ -177,7 +179,8 @@ jobs:
                 ${{ env.cache-name }}-${{ runner.os }}-${{ matrix.ghc }}-
                 ${{ env.cache-name }}-${{ runner.os }}-
 
-      - run: cabal v2-update
+      - run: |
+          cabal v2-update
```

Comment:
not related with this pr but i've just was thinking if `cabal update` should not be before the freeze step, we have the hackage index fixed but freeze would change if the index change (only an observation, no asking for changes)

Reply:
`freeze` seems to save the real Hackage index used. So if `index-state` is pinned to something, the `freeze` infers the real snapshot time before that moment.

That's why it seems that the placement of `update` does not even matter really. It even probably may be removed.

But `update` would play an important role, if `haskell/actions/setup` for some reason has the older Hackage update than the `index-state` package uses. Currently *idk* if `haskell/actions/setup` does `cabal update`:

![Screenshot-2021-12-27-17:06:32](https://user-images.githubusercontent.com/20933385/147484108-1efea2b1-1578-46d9-ae90-0e4056f3eb73.png)

so action reports nothing.

Currently, it does does the update: https://github.com/haskell/actions/blob/a64ac8cca0d4aec4e3fd4a5fbc9ae3c4953e20aa/setup/src/setup-haskell.ts#L75 (well it should update at every run).